policy/lib/rule_data.rego

```diff
@@ -93,19 +95,35 @@ rule_data_defaults := {
 rule_data(key_name) := value if {
 	# Expected to be defined under `configuration.rule_data` in the
 	# ECP configuration data being used when EC is run.
-	value := data.rule_data__configuration__[key_name]
+	value := rule_data_transform(data.rule_data__configuration__, key_name)
 } else := value if {
 	# Expected to be defined in a users custom data source accessed
 	# via an oci bundle or (more likely) a git url.
-	value := data.rule_data_custom[key_name]
+	value := rule_data_transform(data.rule_data_custom, key_name)
 } else := value if {
 	# Expected to be defined in a default data source accessed via
 	# an oci bundle or a maybe a git url. See example/data/rule_data.yml.
-	value := data.rule_data[key_name]
+	value := rule_data_transform(data.rule_data, key_name)
 } else := value if {
 	# Default values defined in this file. See above.
-	value := rule_data_defaults[key_name]
+	value := rule_data_transform(rule_data_defaults, key_name)
 } else := value if {
 	# If the key is not found, default to an empty list
 	value := []
 }
+
+rule_data_transform(rule_data, key) := value if {
```

Comment:
I don't fully understand this. It looks like in the first block we are pulling the value from the top item of the list of acceptable items. While on the second block, we return the list of acceptable items. Can you elaborate on why this is being done? A comment in the code would be super helpful.

My instinct is that this is going to make it much harder to use the rule_data function since the caller now needs to worry about the type of the expected data and how to interpret each one.

Also keep in mind that we have several policy rules that check the schema of the data being used. Most of them will not allow using effective_on.